add documentation for GitHub Actions under CI best practices (#1087)
fastlane/docs: docs/best-practices/continuous-integration/github.md

# GitHub Actions Integration

> Use [GitHub Actions runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) running on a macOS machine
> to build using fastlane.

Reviewer: I think we don't break lines in our docs, so I'd join them:

```suggestion
Use [GitHub Actions runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners) running on a macOS machine to build using _fastlane_.
```

(I also italicized `fastlane` 😁 )

## Repository setup

First create a `Gemfile` in the root of your project with the following content:

```ruby
source 'https://rubygems.org'

gem 'fastlane'
```

> Add a `.github/workflows/build-ios-app.yml` file to trigger _fastlane_.

Reviewer: Let's give it a hint that `build-ios-app` is just a example name the user can modify 😃 

```suggestion
Add a workflow yaml file to your repository. For example, if naming your workflow "build-ios-app", add `.github/workflows/build-ios-app.yml` with the following content:
```

```yml
name: build-ios-app
on:
  push:
    branches:
      - 'master'
jobs:
  build:
    runs-on: macOS-latest
    steps:
      - uses: actions/checkout@v2
      - run: fastlane beta
        env:
          MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
```

> See [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
> for more information on how this file works.

Reviewer: Same here about line break:

```suggestion
See [Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions) for more information on how this file works.
```

## Setting up the lanes

> ```ruby

Reviewer: Again, I think it's worth mentioning this is just a sample fastfile config

```suggestion
Here's an example of a `Fastfile` with a lane that runs _match_, builds the app, and uploads to TestFlight:

```ruby
```

platform :ios do
  lane :beta do
    setup_ci if ENV['CI']
    match(type: 'appstore')
    build_app
    upload_to_testflight(skip_waiting_for_build_processing: true)
  end
end
```

> `setup_ci` creates a temporary keychain. Without this, the build could freeze and never finish.

Reviewer: I think this would be better worded like this:

```suggestion
Note the usage of `setup_ci`: it creates a temporary keychain. Without this, the build could freeze and never finish.
```
